feat: 端口转发功能

endymx opened the pull request:
#### What this PR does / why we need it?
* 增加 ufw/firewalld 防火墙下对端口转发的支持（含前端页面）
* 考虑到 ufw 本身转发会导致重启服务时多次添加相同规则（曾经使用宝塔遭遇过该问题）与实现难度，改为使用iptables实现转发

#### Summary of your change
**@endymx**
* 后端api实现
* 增加1panel启动时初始化端口转发规则：
  (ufw) 清空iptables nat表规则，根据数据库内规则reload
  (firewalld) 检测端口伪装并开启
* ufw 下增删转发规则时，同步修改数据库
* 测试功能在 Debian12 与 Centos7 下的可用性

**@yawntee**
* 前端页面实现

**@ExerciseBook**
* 需求跟进

#### Please indicate you've done the following:
- [x] Made sure tests are passing and test coverage is added if needed.
- [x] Made sure commit message follow the rule of [Conventional Commits specification](https://www.conventionalcommits.org/).
- [x] Considered the docs impact and opened a new docs issue or PR with docs changes if needed.

endymx committed:
feat: 端口转发功能


Co-authored-by: yawntee <yawntee@qq.com>

wanghe-fit2cloud commented:
感谢支持。

ssongliu commented on frontend/src/views/host/firewall/forward/index.vue:
```diff
@@ -0,0 +1,255 @@
+                            <el-table-column :label="$t('commons.table.protocol')" :min-width="70" prop="protocol" />
+                            <el-table-column :label="$t('firewall.sourcePort')" :min-width="70" prop="port" />
+                            <!-- <el-table-column :label="$t('commons.table.status')" :min-width="120">
+                                <template #default="{ row }">
```
删除掉注释代码

ssongliu commented on frontend/src/views/host/firewall/forward/index.vue:
```diff
@@ -0,0 +1,255 @@
+const data = ref();
+const paginationConfig = reactive({
+    cacheSizeKey: 'firewall-port-page-size',
+    currentPage: 1,
```
forward-page-size

ssongliu commented on frontend/src/views/host/firewall/forward/index.vue:
```diff
@@ -0,0 +1,255 @@
+            data.value = res.data.items || [];
+            for (const item of data.value) {
+                item.usedPorts = item.usedStatus ? item.usedStatus.split(',') : [];
+            }
```
转发端口后端没有走 usedPorts 这部分，可以直接去掉

ssongliu commented on frontend/src/views/host/firewall/forward/operate/index.vue:
```diff
@@ -0,0 +1,174 @@
+
+const rules = reactive({
+    protocol: [Rules.requiredSelect],
+    port: [Rules.requiredInput],
```
port 的规则可以直接用 Rules.port ，因为端口那边支持端口范围，所以没法直接用，这里可以用

endymx replied:
考虑到之后会有范围端口转发的需求，还是采用原有方案

ssongliu commented on backend/app/service/firewall.go:
```diff
@@ -300,6 +296,28 @@ func (u *FirewallService) OperatePortRule(req dto.PortRuleOperate, reload bool)
 }
 
+func (u *FirewallService) OperateForwardRule(req dto.ForwardRuleOperate) error {
+	client, err := firewall.NewFirewallClient()
```
这里需要注意一下批量删除方法，测试的时候，添加 8080 - 80 8080 - 81 两条规则，然后批量删除会报错，因为删除规则时，num 会变化

ssongliu commented on backend/router/ro_host.go:
```diff
@@ -29,6 +29,7 @@ func (s *HostRouter) InitRouter(Router *gin.RouterGroup) {
 		hostRouter.POST("/firewall/search", baseApi.SearchFirewallRule)
 		hostRouter.POST("/firewall/operate", baseApi.OperateFirewall)
 		hostRouter.POST("/firewall/port", baseApi.OperatePortRule)
+		hostRouter.PUT("/firewall/forward", baseApi.OperateForwardRule)
```
可以直接改成 post 方法，1panel 里面基本没有用 PUT

ssongliu commented:
非常感谢代码贡献，有一点点地方可以优化一下。

ExerciseBook committed:
fix: cr


refactor(forward): 优化前端代码
fix: 删除iptables规则时排序问题
fix: router注释遗漏
refactor(forward): 优化i18n描述
fix(forward): 修复参数校验
fix(forward): ip可空
fix: 删除debug部分

Co-authored-by: endymx <345793738@qq.com>
Co-authored-by: yawntee <yawntee@qq.com>
